framework/include/utils/MooseTypes.h

```diff
@@ -342,6 +344,11 @@ struct ADType<Real>
   typedef ADReal type;
 };
 template <>
+struct ADType<ChainedReal>
```

Comment:
we have two structs doing the same thing. The other one is `MooseADWrapper`

Reply:
Okay. I added that one in MooseADWrapper.h as well.